app/controller/command/commands/export.py

```diff
@@ -0,0 +1,183 @@
+        parser_view.add_argument("--team", metavar="TEAM",
+                                 type=str, action='store',
+                                 help="(Admin/Lead only) Export emails"
+                                      " by Team Name")
+
+        return subparsers
+
+    def get_help(self, subcommand: str = None) -> str:
+        """Return command options for user events with Slack formatting."""
+
+        def get_subcommand_help(sc: str) -> str:
+            """Return the help message of a specific subcommand."""
+            message = f"\n*{sc.capitalize()}*\n"
+            message += self.subparser.choices[sc].format_help()
+            return message
+
+        if subcommand is None or subcommand not in self.subparser.choices:
+            res = f"\n*{self.command_name} commands:*```"
+            for argument in self.subparser.choices:
+                res += get_subcommand_help(argument)
+            return res + "```"
+        else:
+            res = "\n```"
+            res += get_subcommand_help(subcommand)
+            return res + "```"
+
+    def handle(self,
+               command: str,
+               user_id: str) -> ResponseTuple:
+        """Handle command by splitting into substrings and giving to parser."""
+        logging.debug("Handling ExportCommand")
+        command_arg = shlex.split(command)
+        args = None
+
+        try:
+            args = self.parser.parse_args(command_arg)
+        except SystemExit:
+            all_subcommands = list(self.subparser.choices.keys())
+            present_subcommands = [subcommand for subcommand in
+                                   all_subcommands
+                                   if subcommand in command_arg]
+            present_subcommand = None
+            if len(present_subcommands) == 1:
+                present_subcommand = present_subcommands[0]
+            return self.get_help(subcommand=present_subcommand), 200
+
+        if args.which == "emails":
+            try:
+                command_user = self.facade.retrieve(User, user_id)
+                if not check_permissions(command_user, None):
+                    return self.permission_error, 200
+
+                # Check if team name is provided
+                if args.team is not None:
+                    users = self.get_team_users(args.team)
+                    return self.export_emails_helper(users)
+                else:  # if team name is not provided, export all emails
+                    users = self.facade.query(User)
+                    return self.export_emails_helper(users)
+            except LookupError:
+                return self.lookup_error, 200
+        else:
+            return self.get_help(), 200
+
+    def get_team_users(self, team_name):
+        team = get_team_by_name(self.facade, team_name)
+
+        params = [('github_user_id', ids)
+                  for ids in list(team.members)]
+        return self.facade.query_or(User, params)
```

Comment:
Now, there's this other function called [`get_team_members()`](https://rocket2.readthedocs.io/en/latest/docs/api/db.html#db.utils.get_team_members)....